File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        // Range of Answer : [0, n]
        int lt=0, rt=nums.size()-1, mid;
        while(lt<=rt){
            mid = lt+(rt-lt)/2;
            nums[mid]<target?lt=mid+1:rt=mid-1;
        }
        return lt;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int lt = 0, rt = nums.length - 1;
        while (lt <= rt) {
            int mid = (lt + rt) / 2;
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] < target) {
                lt = mid + 1;
            } else {
                rt = mid - 1;
            }
        }
        return lt;
    }
}